### PR TITLE
feat: BitVec.zero_sub

### PR DESCRIPTION
**Pull request by eurquhart1**

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
Wow, lovely. I guess you may want to generalize this to `w` instead of `1`. Look into `simp [bv_toNat]; omega` or directly `bv_omega`.

File: SSA/Projects/InstCombine/ForLean.lean [hunk 1]
```diff
@@ -621,6 +621,10 @@ theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} (n : Nat) :
     BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
   simp [← BitVec.shiftLeft_or_distrib]
 
+@[simp]
+theorem neg_eq_sub_zero (x : BitVec w) : - x = 0 - x := by
+    simp [bv_toNat]
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/ForLean.lean]:
```suggestion
theorem zero_sub {x : BitVec w} : 0 - x = - x = := by
    simp [bv_toNat]
```

Comment (tobiasgrosser) [reply SSA/Projects/InstCombine/ForLean.lean]: We typically want the simper variant on the left. And we can name only the LHS if the RHS is obvious.

Comment (tobiasgrosser) [reply SSA/Projects/InstCombine/ForLean.lean]: Also for equalities where x arises on both sides, we can use implicit arguments.

**Review by tobiasgrosser — APPROVED** [review]
Feel free to merge after accepting my comment.

File: SSA/Projects/InstCombine/ForLean.lean [hunk 2]
```diff
@@ -621,6 +621,10 @@ theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} (n : Nat) :
     BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
   simp [← BitVec.shiftLeft_or_distrib]
 
+@[simp]
+theorem zero_sub (x : BitVec w) : 0 - x = - x = := by
+    simp [bv_toNat]
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/ForLean.lean]:
```suggestion
theorem zero_sub {x : BitVec w} : 0 - x = - x := by
    simp [bv_toNat]
```

Comment (tobiasgrosser) [reply SSA/Projects/InstCombine/ForLean.lean]: My mistake.

File: SSA/Projects/InstCombine/ForLean.lean [hunk 3]
```diff
@@ -621,6 +621,10 @@ theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} (n : Nat) :
     BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
   simp [← BitVec.shiftLeft_or_distrib]
 
+@[simp]
+theorem zero_sub {x : BitVec w} : 0 - x = - x := by
+    simp [bv_toNat]
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/ForLean.lean]:
```suggestion
theorem zero_sub {x : BitVec w} : 0#w - x = - x := by
  simp [bv_toNat]
```

And we probably want to have the simp-normal-form for zero bitvectors, where the bitwidth is explicit. And the proof is typically indented by 2.

Comment (eurquhart1) [reply SSA/Projects/InstCombine/ForLean.lean]: thanks for the comments!

**Comment by tobiasgrosser** [comment]
Feel free to merge.

**Pull request merged**